ipatests/pytest_ipa/integration/__init__.py

```diff
@@ -477,15 +510,26 @@ def add_compat_attrs(cls, mh):
     This is deprecated in favor of the mh fixture.
     To be removed when no more tests using this.
     """
-    cls.domain = mh.domain
-    cls.master = mh.master
-    cls.replicas = mh.replicas
-    cls.clients = mh.clients
-    cls.ad_domains = mh.config.ad_domains
-    if cls.ad_domains:
-        cls.ads = mh.ads
-        cls.ad_subdomains = mh.ad_subdomains
-        cls.ad_treedomains = mh.ad_treedomains
+    if cls.num_trusted_domains == 1:
```

Comment:
Instead of having 2 separate code branches, you can keep the old structure and add
```
    cls.trusted_domains = mh.config.trusted_domains
    if cls.trusted_domains:
        cls.trusted_master = mh.trusted_master
        cls.trusted_replicas = mh.trusted_replicas
        cls.trusted_clients = mh.trusted_clients
```